ownCloudAppFramework/DisplaySettings.m

```diff
@@ -0,0 +1,144 @@
+//
+//  DisplaySettings.m
+//  ownCloud
+//
+//  Created by Felix Schwarz on 21.05.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2019, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+#import "DisplaySettings.h"
+
+@implementation DisplaySettings
+
+#pragma mark - Class settings
++ (OCClassSettingsIdentifier)classSettingsIdentifier
+{
+	return (OCClassSettingsIdentifierDisplay);
+}
+
++ (NSDictionary<OCClassSettingsKey,id> *)defaultSettingsForIdentifier:(OCClassSettingsIdentifier)identifier
+{
+	if ([identifier isEqual:OCClassSettingsIdentifierDisplay])
+	{
+		return (@{
+			OCClassSettingsKeyDisplayShowHiddenFiles : @(NO)
+		});
+	}
+
+	return (nil);
+}
+
+#pragma mark - Singleton
++ (DisplaySettings *)sharedDisplaySettings
```

Comment:
@felix-schwarz wouldn't it be better to call it just shared, since `[DisplaySesttings sharedDisplaySettings]` is too explicit for my taste at least?

Reply:
This is to match the naming conventions (i.e. `NSWorkspace.sharedWorkspace`, `NSAppleEventManager.sharedAppleEventManager`, etc.). It will be translated to `DisplaySettings.shared` in Swift automatically.